lib/rules/require-default-props.js

```diff
@@ -87,6 +112,65 @@ module.exports = {
       });
     }
 
+    /**
+     * If functions option is 'defaultArguments', reports defaultProps is used and all params that doesn't initialized.
+     * @param {Object} componentNode Node of component.
+     * @param {Object[]} declaredPropTypes List of propTypes to check `isRequired`.
+     * @param {Object} defaultProps Object of defaultProps to check used.
+     */
+    function reportFunctionComponent(componentNode, declaredPropTypes, defaultProps) {
+      if (defaultProps) {
+        report(context, messages.noDefaultPropsWithFunction, 'noDefaultPropsWithFunction', {
+          node: componentNode,
+        });
+      }
+      if (componentNode.params.length === 0) {
+        return;
```

Comment:
this can be covered with a functional component that takes no props, no?

Reply:
[`f38ffa8` (#3249)](https://github.com/jsx-eslint/eslint-plugin-react/pull/3249/commits/f38ffa80909830fb5945039f7d73d7b1fbf14c2f)
I think I cover it above. But I keep getting warnings. What should I do?

Reply:
Oh sorry. Now I see it has been fixed.

Reply:
it doesn't seem to be fixed?

Reply:
https://github.com/jsx-eslint/eslint-plugin-react/pull/3249/commits/f38ffa80909830fb5945039f7d73d7b1fbf14c2f
Oh... but I coverd it in the commit above. What's wrong?

Reply:
Looks like you already filtered out everything that has no declared propTypes in the Program:exit listener, so it never gets here with those test cases.

Removing that filtering, and adding a couple additional guards, covers the line - but i don't think that's necessary, so I'll just remove it.